"""Protobuf schema for compliance logs."""

from lokryn_mcp_log.schema.log_pb2 import (
    LogRequest,
    AppendResponse,
    StreamAck,
    Severity,
    Outcome,
    EventType,
    Sensitivity,
    SEVERITY_UNSPECIFIED,
    SEVERITY_DEBUG,
    SEVERITY_INFO,
    SEVERITY_NOTICE,
    SEVERITY_WARNING,
    SEVERITY_ERROR,
    SEVERITY_CRITICAL,
    SEVERITY_ALERT,
    SEVERITY_EMERGENCY,
    OUTCOME_UNSPECIFIED,
    OUTCOME_SUCCESS,
    OUTCOME_FAILURE_UNAUTHORIZED,
    OUTCOME_FAILURE_DENIED,
    OUTCOME_FAILURE_ERROR,
    OUTCOME_PARTIAL,
    EVENT_TYPE_UNSPECIFIED,
    EVENT_LOGIN,
    EVENT_LOGOUT,
    EVENT_FILE_ACCESS,
    EVENT_POLICY_CHANGE,
    EVENT_PRIVILEGE_USE,
    EVENT_CONFIG_CHANGE,
    EVENT_DATA_EXPORT,
    EVENT_NETWORK_CONNECTION,
    EVENT_PROCESS_START,
    EVENT_PROCESS_STOP,
    EVENT_USER_MANAGEMENT,
    EVENT_RESOURCE_ACCESS,
    EVENT_TOOL_INVOCATION,
    EVENT_MODEL_INFERENCE,
    EVENT_AGENT_DECISION,
    EVENT_DELEGATION,
    EVENT_CONTEXT_ACCESS,
    EVENT_PROMPT_EXECUTION,
    EVENT_GUARDRAIL_CHECK,
    SENSITIVITY_UNSPECIFIED,
    SENSITIVITY_PUBLIC,
    SENSITIVITY_INTERNAL,
    SENSITIVITY_CONFIDENTIAL,
    SENSITIVITY_RESTRICTED,
    SENSITIVITY_HIGHLY_RESTRICTED,
)

__all__ = [
    "LogRequest",
    "AppendResponse",
    "StreamAck",
    "Severity",
    "Outcome",
    "EventType",
    "Sensitivity",
    "SEVERITY_UNSPECIFIED",
    "SEVERITY_DEBUG",
    "SEVERITY_INFO",
    "SEVERITY_NOTICE",
    "SEVERITY_WARNING",
    "SEVERITY_ERROR",
    "SEVERITY_CRITICAL",
    "SEVERITY_ALERT",
    "SEVERITY_EMERGENCY",
    "OUTCOME_UNSPECIFIED",
    "OUTCOME_SUCCESS",
    "OUTCOME_FAILURE_UNAUTHORIZED",
    "OUTCOME_FAILURE_DENIED",
    "OUTCOME_FAILURE_ERROR",
    "OUTCOME_PARTIAL",
    "EVENT_TYPE_UNSPECIFIED",
    "EVENT_LOGIN",
    "EVENT_LOGOUT",
    "EVENT_FILE_ACCESS",
    "EVENT_POLICY_CHANGE",
    "EVENT_PRIVILEGE_USE",
    "EVENT_CONFIG_CHANGE",
    "EVENT_DATA_EXPORT",
    "EVENT_NETWORK_CONNECTION",
    "EVENT_PROCESS_START",
    "EVENT_PROCESS_STOP",
    "EVENT_USER_MANAGEMENT",
    "EVENT_RESOURCE_ACCESS",
    "EVENT_TOOL_INVOCATION",
    "EVENT_MODEL_INFERENCE",
    "EVENT_AGENT_DECISION",
    "EVENT_DELEGATION",
    "EVENT_CONTEXT_ACCESS",
    "EVENT_PROMPT_EXECUTION",
    "EVENT_GUARDRAIL_CHECK",
    "SENSITIVITY_UNSPECIFIED",
    "SENSITIVITY_PUBLIC",
    "SENSITIVITY_INTERNAL",
    "SENSITIVITY_CONFIDENTIAL",
    "SENSITIVITY_RESTRICTED",
    "SENSITIVITY_HIGHLY_RESTRICTED",
]
